### DISCUSSION TITLE
CoPilot works poorly with Python

### DISCUSSION THREAD

**Discussion by joakimsii** [community/community#13620] [Copilot Conversations] [answered]
For example "#ask the user for their age" responses 
```
#if they are 18 or older, print "you can vote"
#if they are 18 or younger, print "you can't vote"
```
Another example is this
```
#randomize an number
import random
```
As you can see, sometimes it just print's how the program should go forward and sometimes it doesn't end the script, and sometimes it doesn't do nothing. I thin that Copilot need much more training on python, as Python is so big of an language

**Comment by Naiefhq** [comment, marked as answer]
Hi 

From what i observed so far is that copilot does not write a program on a given comment. It needed some user input. I took your comments above pasted it and then 

def main()

that gave me suggested solutions. :)

#if they are 18 or older, print "you can vote"
#if they are 18 or younger, print "you can't vote"

def main():
    age = int(input("How old are you? "))
    if age >= 18:
        print("You can vote")
    else:
        print("You can't vote")

if __name__ == "__main__":
    main()


#randomize an number

def main():
    import random
    print("This program generates a random number between 1 and 100.")
    print("The number is:", random.randint(1,100))

if __name__ == "__main__":
    main()
